draft-ietf-jsonpath-base.md

```diff
@@ -1068,22 +1068,22 @@ Note that there is no "bald" descendant selector `..`.
 #### Semantics
 {: unnumbered}
 
-The `descendant-selector` selects certain descendants of a node:
+A `descendant-selector` selects certain descendants of a node:
 
 * the `..<name>` form (and the `..[<index>]` form where `<index>` is a `quoted-member-name`) selects those descendants that are member values of an object with the given member name.
 * the `..[<index>]` form, where `<index>` is an `element-index`, selects those descendants that are array elements with the given index.
 * the `..[<slice-index>]` form selects those descendants that are array elements selected by the given slice.
 * the `..[<filter>]` form selects those descendants that are array elements or object values selected by the given filter.
 * the `..[*]` and `..*` forms select all the descendants.
 
-The resultant nodelist is ordered as if:
+An _array preorder_ of the descendants of a node is a sequence of all the descendants in which:
```

Comment:
Hmm, the "preorder" part is universal and not just for arrays.  So qualifying "preorder" with "array" doesn't strike me as helpful.  Simply calling it "preorder" is fine, but I wanted to add an adjective that indicates that this conventional order is one of many possible (JSON objects), which the implementation is free to choose, and that it is the order we have chosen for JSONPath.

Reply:
Yeah, I was trying to capture an additional constraint on the order: it is both a preorder (parents before children) and arrays are traversed in array order (not implied by preorder).

So I tried to introduce a new term "array preorder" which is a special kind of preorder.

I'm not sure about an adjective conveying the lack of ordering of objects. Even if we found a suitable adjective, we'd still need to build in the array condition, ideally into the term, somehow.

Reply:
After some more thinking, I don't get a better result than "array-ordered pre-order".  Maybe  bit redundant and wordy, but we are not using this term all over the place.

Reply:
That's certainly explicit, but the ordered/order part doesn't sound good. Do any of the following grab you?

* array-based preorder
* arraywise preorder
* array-respecting preorder
* arrayful preorder
* linear preorder
* vector preorder

Reply:
Surprisingly hard.

* array-sequenced pre-order
* array-guided pre-order

Reply:
"array-sequenced preorder" seems to be the best term so far. (I think preorder is usually unhyphenated. See, for example, [wikipedia](https://en.wikipedia.org/wiki/Preorder).)